### PR TITLE
Fix regex page matcher for Next.js middlewares version 1

### PR DESCRIPTION
**Pull request by leoortizz**
### Description

Projects using the first version of the Next.js middleware were failing to deploy due to a difference in `middleware-manifest.json`. The first version was in Next.js until [v12.1.6](https://github.com/vercel/next.js/commit/b188fab3360855c28fd9407bd07c4ee9f5de16a6). Version 2 was introduced in [Next.js 12.2.0](https://nextjs.org/docs/messages/middleware-upgrade-guide). This PR handles the middleware page matchers for version 1.

### CODE REVIEW

File: src/frameworks/next/index.ts [hunk 1]
```diff
@@ -277,13 +277,19 @@ export async function ɵcodegenPublicDirectory(sourceDir: string, destDir: strin
   }
 
   const [middlewareManifest, prerenderManifest, routesManifest] = await Promise.all([
-    readJSON<MiddlewareManifest>(join(sourceDir, distDir, "server", MIDDLEWARE_MANIFEST)),
+    readJSON<MiddlewareManifest | MiddlewareManifestV1>(
```
Comment (jamesdaniels) [inline src/frameworks/next/index.ts:280]: can we instead make `type MiddlewareManifest = MiddlewareManifestV1 | MiddlewareManifestV2`

Comment (jamesdaniels) [reply src/frameworks/next/index.ts:280]: alt, if the middleware manifest type is exported by Next 12 we could add next 12 as a named devDep in the package.json and leverage that. e.g, `"next12": "npm:next@^12", "next13": "npm:next@^13"`

**Review by jamesdaniels — CHANGES_REQUESTED** [review]

**Comment by jamesdaniels** [comment]
Are there any other Next 12 gotchas? E.g, rewrites/redirects/headers?

File: src/frameworks/next/index.ts (hunk 1 above)
Comment (leoortizz) [reply src/frameworks/next/index.ts:280]: I initially tried having `next12` as a separate dev dep but Next.js has some internal imports from `next/dist/...` that breaks the build. I can investigate further

**Comment by leoortizz** [comment]
> Are there any other Next 12 gotchas? E.g, rewrites/redirects/headers?

I can check but I don't think so. Middlewares was still in beta before Next.js 12.2 so that was the main thing we're dealing with that has changed

File: src/frameworks/next/index.ts (hunk 1 above)
Comment (jamesdaniels) [reply src/frameworks/next/index.ts:280]: @leoortizz if you tried that and it didn't work well, let's punt. Not worth the time for Next 12, which we'll probably drop at some point

**Comment by jamesdaniels** [comment]
Add changelog

File: src/frameworks/next/index.ts [hunk 2]
```diff
@@ -288,7 +287,9 @@ export async function ɵcodegenPublicDirectory(sourceDir: string, destDir: strin
   ]);
 
   const middlewareMatcherRegexes = Object.values(middlewareManifest.middleware)
-    .map((it) => it.matchers)
+    .map((page: MiddlewareManifest["middleware"]["page"]) =>
```
Comment (TheIronDev) [inline src/frameworks/next/index.ts:290]: Test this branch? (bonus points illustrating a test -case a previously broken case and showing it works now.)

**Review by TheIronDev — CHANGES_REQUESTED** [review]
Please write a test to "prove" this addresses the mismatch middleware issue.

**Review by jamesdaniels — APPROVED** [review]

**Pull request merged**